### DISCUSSION TITLE
[blocks]: Double sidebar panel with collapsible all menu panel

### DISCUSSION THREAD

**Discussion by adiramardiani** [shadcn-ui/ui#3167] [Blocks Request]
### Description

Hi @shadcn team

I see many people want to admin template, and refer to ant design, for collapsible sidebar
Here my suggest about sidebar, but this time is double sidebar, I think if this can implement, the basic sidebar will developed too

I'm still looking for the right basic block template using shadcn, there is still no mode that has dual menu/two menu/etc, whatever any one call this.

Currently I am focusing on ease of navigation for the user, where he can move to pages he has previously opened or in the same context quickly.

https://github.com/shadcn-ui/ui/assets/8703419/24e5678e-dc5d-4c90-9f3d-f21219b9f84a

I refer to vben admin, this template is free and using ant design vue as base
- The navigation header on top is fix, so if content scroll, the header is still
- With the mini menu on the side, applications that have lots of menus can be made neater and more organized
- Can also be used to store main modules, such as finance, accounting, warehouse, etc
- The mini menu is scrollable, and can change mode from icon only (and tooltip) to icon with text
- The other menu can be pinned, so we can make the menu remain there, or make it minimal
- So if the mini menu or main menu have many item, still have beautiful scroll using Scroll-area
- I think it's a good idea if the shadcn have this block ❤️ 
- currently I not see any example or block to build sidebar with nested menu, I don't know using shadcn accordion, or combining something else to support nested level to 3 level

This is optional
- The `tab` mechanism for previously opened pages allows users to move very quickly, `so so good idea`
- I try make MANY tab, and still organized using scroll and next/prev button
- a right-click feature to bring up options from a tab would also be very helpful (perfect to combine with Context Menu)
- breadcrumb menu is matching with shadcn component with dropdown ability
- mini menu can complete hide, so all menu is hidden, and can show using something like this, but the purpose of button is to hide/show mini menu:

https://github.com/shadcn-ui/ui/assets/8703419/579e5224-4820-485c-af23-0d5c184b9625

We had something similiar, but I thing the admin case will match to many user need as base template
![image](https://github.com/shadcn-ui/ui/assets/8703419/8a03b1da-ccbb-4410-869d-20aa39aeced7)

I think if shadcn had a menu model like this would be amazing, and implementation about `Shadcn Block` will more hype 👍 

This is the link for vben admin https://vben.vvbin.cn

### Example

https://vben.vvbin.cn

**Comment by meimosor** [comment]
+1

**Comment by rajanmodi30** [comment]
here is a good one you guys can try that i found github - https://github.com/gaofubin/t3-app-template , live - https://t3-app-template.vercel.app/

**Reply by ofoetey** [reply]
@rajanmodi30 it works fine and perfect, had to do some small customizations

**Reply by NickWoodward** [reply]
I don't think this is what he's after? Unless I've missed the double sidebar on this live demo?

**Comment by NickWoodward** [comment]
> ### Description
> 
> Hi @shadcn team
> 
> I see many people want to admin template, and refer to ant design, for collapsible sidebar Here my suggest about sidebar, but this time is double sidebar, I think if this can implement, the basic sidebar will developed too
> 
> I'm still looking for the right basic block template using shadcn, there is still no mode that has dual menu/two menu/etc, whatever any one call this.
> 
> Currently I am focusing on ease of navigation for the user, where he can move to pages he has previously opened or in the same context quickly.
> Vben.Recording.2024-03-23.204852.mp4
> 
> I refer to vben admin, this template is free and using ant design vue as base
> 
>     * The navigation header on top is fix, so if content scroll, the header is still
> 
>     * With the mini menu on the side, applications that have lots of menus can be made neater and more organized
> 
>     * Can also be used to store main modules, such as finance, accounting, warehouse, etc
> 
>     * The mini menu is scrollable, and can change mode from icon only (and tooltip) to icon with text
> 
>     * The other menu can be pinned, so we can make the menu remain there, or make it minimal
> 
>     * So if the mini menu or main menu have many item, still have beautiful scroll using Scroll-area
> 
>     * I think it's a good idea if the shadcn have this block ❤️
> 
>     * currently I not see any example or block to build sidebar with nested menu, I don't know using shadcn accordion, or combining something else to support nested level to 3 level
> 
> 
> This is optional
> 
>     * The `tab` mechanism for previously opened pages allows users to move very quickly, `so so good idea`
> 
>     * I try make MANY tab, and still organized using scroll and next/prev button
> 
>     * a right-click feature to bring up options from a tab would also be very helpful (perfect to combine with Context Menu)
> 
>     * breadcrumb menu is matching with shadcn component with dropdown ability
> 
>     * mini menu can complete hide, so all menu is hidden, and can show using something like this, but the purpose of button is to hide/show mini menu:
> 
> 
> Hide.mini.menu.Recording.2024-03-23.210310.mp4
> 
> We had something similiar, but I thing the admin case will match to many user need as base template ![image](https://private-user-images.githubusercontent.com/8703419/316246455-8a03b1da-ccbb-4410-869d-20aa39aeced7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3OTAzMjMsIm5iZiI6MTczMzc5MDAyMywicGF0aCI6Ii84NzAzNDE5LzMxNjI0NjQ1NS04YTAzYjFkYS1jY2JiLTQ0MTAtODY5ZC0yMGFhMzlhZWNlZDcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIxMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMTBUMDAyMDIzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDE3OWFmMzNkZTk2ZTUwOGE4MmNkOTY3OGFjODRkNWVmMzdjMWNmZDNiYTRjZTM3MzJiMGY5MGU1Y2FlMWU2NiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.cg5VeMAGwI8ym1j0dr5DZNtRd3dHjTsinWnihOR_ovE)
> 
> I think if shadcn had a menu model like this would be amazing, and implementation about `Shadcn Block` will more hype 👍
> 
> This is the link for vben admin https://vben.vvbin.cn
> ### Example
> 
> https://vben.vvbin.cn

did you get anywhere with this?

blocks now has a double menu, but the rhs is fixed, presumably because the provider is used for state.

I've managed to wrap both menus in their own provider, with their own cookies for state, but I struggle with content between the sliders. The left acts as I want, pushing content, but the right hand menu will only overlay.
